test/DataArrayTest.cpp

```diff
@@ -21,6 +22,38 @@ TEST_CASE("DataArrayCreation")
   REQUIRE(numTuples == 0);
 }
 
+TEST_CASE("complex::DataArray Copy TupleTest", "[complex][DataArray]")
+{
+  DataStructure dataStructure;
+  IDataStore::ShapeType tupleShape{5};
+  IDataStore::ShapeType componentShape{3};
+  Result<> result = CreateArray<int32>(dataStructure, tupleShape, componentShape, DataPath({"DataArray"}), IDataAction::Mode::Execute);
+  DataArray<int32>& dataArray = ArrayRefFromPath<int32>(dataStructure, DataPath({"DataArray"}));
+
+  for(size_t i = 0; i < 5; i++)
```

Comment:
The loops should use the values from the tuple/component shapes instead of being hardcoded.